### PR TITLE
fix: infinity spinner when no post or comments on profile screen.

### PR DESCRIPTION
**Pull request by qoopa**
ListEmptyComponent had a spinner so when no result where found it would have kept spinning until end of time.

Change to display no comments message.

Made a `NoResultView` to share style for `NoCommensView` and `NoPostsView`

### CODE REVIEW

File: components/ui/common/NoResultView.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+export interface INoResultViewProps extends InterfaceVStackProps {
+  message?: string;
+}
+
+function NoResultView({ message, ...rest }: INoResultViewProps) {
+  const theme = useTheme();
+
+  return (
+    <VStack flex={1} justifyContent="center" alignItems="center" {...rest}>
```
Comment (sgriff96) [inline components/ui/common/NoResultView.tsx:13]: I think instead we could give this a `type` that would be like `comment` or `post` and return a different message based on that. And if no type is provided we just return a default message "No results found :("

**Review by sgriff96 — CHANGES_REQUESTED** [review]
Left a comment, but this is good thanks for helping!

File: components/ui/comments/NoCommentsView.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from "react";
+import NoResultView, { INoResultViewProps } from "../common/NoResultView";
+
+function NoCommensView({ ...props }: INoResultViewProps) {
```
Comment (sgriff96) [inline components/ui/comments/NoCommentsView.tsx:4]: typo here, `Commens` instead of `Comments`

File: components/screens/post/PostScreen.tsx [hunk 3]
```diff
@@ -140,13 +140,7 @@ function PostScreen({
...
     }
     if (post.comments.length === 0 && !post.commentsError) {
-      return (
-        <Center my={4}>
-          <Text fontStyle="italic" color={theme.colors.app.textSecondary}>
-            No comments yet. Time to do your part 🫡
-          </Text>
-        </Center>
-      );
+      return <NoCommensView my={4} />;
```
Comment (sgriff96) [inline components/screens/post/PostScreen.tsx:143]:
```suggestion
      return <NoCommentsView my={4} />;
```

File: components/screens/post/PostScreen.tsx [hunk 4]
```diff
@@ -25,6 +24,7 @@ import ContentView from "../../ui/ContentView";
 import LoadingErrorFooter from "../../ui/Loading/LoadingErrorFooter";
 import LoadingView from "../../ui/Loading/LoadingView";
 import PostActionBar from "./PostActionBar";
+import NoCommensView from "../../ui/comments/NoCommentsView";
```
Comment (sgriff96) [inline components/screens/post/PostScreen.tsx:27]:
```suggestion
import NoCommentsView from "../../ui/comments/NoCommentsView";
```

File: components/screens/userProfile/UserProfileScreen.tsx [hunk 5]
```diff
@@ -35,6 +27,8 @@ import CommentItem from "../../ui/comments/CommentItem";
 import LoadingErrorView from "../../ui/Loading/LoadingErrorView";
 import LoadingView from "../../ui/Loading/LoadingView";
 import NotFoundView from "../../ui/Loading/NotFoundView";
+import NoPostsView from "../../ui/Feed/NoPostsView";
+import NoCommensView from "../../ui/comments/NoCommentsView";
```
Comment (sgriff96) [inline components/screens/userProfile/UserProfileScreen.tsx:31]:
```suggestion
import NoCommentsView from "../../ui/comments/NoCommentsView";
```

File: components/screens/userProfile/UserProfileScreen.tsx [hunk 6]
```diff
@@ -225,7 +219,7 @@ function UserProfileScreen({
         estimatedItemSize={100}
         data={profile.comments}
         keyExtractor={commentKeyExtractor}
-        ListEmptyComponent={<Spinner pt={10} />}
+        ListEmptyComponent={<NoCommensView />}
```
Comment (sgriff96) [inline components/screens/userProfile/UserProfileScreen.tsx:222]:
```suggestion
        ListEmptyComponent={<NoCommentsView />}
```

File: components/ui/comments/NoCommentsView.tsx (hunk 2 above)
Comment (sgriff96) [inline components/ui/comments/NoCommentsView.tsx:4]:
```suggestion
function NoCommentsView({ ...props }: INoResultViewProps) {
```

File: components/ui/comments/NoCommentsView.tsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  return (
+    <NoResultView
+      message="No comments yet. Time to do your part 🫡"
+      {...props}
+    />
+  );
+}
+
+export default NoCommensView;
```
Comment (sgriff96) [inline components/ui/comments/NoCommentsView.tsx:13]:
```suggestion
export default NoCommentsView;
```

**Review by sgriff96 — COMMENTED** [review]
In case you feel strongly about not doing my first suggestion, also just added these for fixing the comment typo

File: components/ui/common/NoResultView.tsx (hunk 1 above)
Comment (qoopa) [reply components/ui/common/NoResultView.tsx:20]: @sgriff96 Agreed, and fixed

**Comment by gkasdorf** [comment]
Awesome, thanks for the help! Seriously appreciate it.

**Pull request merged**